File: solutions/9/9.java
import java.lang.Math;

class Main {
    public static boolean isInteger(double number) {
        return Math.floor(number) == number;
    }

    public static int specialPythagoreanTriplet(int triplet_sum) {
        for (int a = 1; a < triplet_sum; a++) {
            for (int b = a + 1; b < triplet_sum - a; b++) {
                double c = Math.sqrt(a * a + b * b);

                if (isInteger(c) && (int) (a + b + c) == triplet_sum) {
                    return (int) (a * b * c);
                }
            }
        }

        return -1;
    }

    public static void main(String args[]) {
        int result = specialPythagoreanTriplet(1000);

        System.out.println(result);
    }
}

File: solutions/9/9.rs
fn is_integer(number: f32) -> bool {
    return number.floor() == number;
}

fn special_pythagorean_triplet(triplet_sum: usize) -> usize {
    for a in 1..triplet_sum {
        for b in a+1..triplet_sum-a {
            let c: f32 = ((a * a + b * b) as f32).sqrt();

            if is_integer(c) && a + b + (c as usize) == triplet_sum {
                return a * b * (c as usize);
            }
        }
    }

    return 0;
}

fn main() {
    let result: usize = special_pythagorean_triplet(1000);

    println!("{}", result);
}